script/install.sh

```diff
@@ -76,9 +77,10 @@ download() {
                 log "Falling back to mirror: ${mirrors[$i]}"
             fi
             if curl "${mirrors[$i]}" -o "$path" ; then
-                break
+                return
             fi
         done
+        return 1
     }
     retry try_all_mirrors
```

Comment:
Something to keep in mind is that `if some_function` completely disables `set -e` inside that function.